tremor-connectors-azure/src/monitor/ingest/writer/sink.rs

```diff
@@ -0,0 +1,296 @@
+            let mut correlation_meta = None;
+
+            for (value, meta) in event.value_meta_iter() {
+                match &mut builder {
+                    None => {
+                        // take the metadata from the first element of the batch
+                        correlation_meta = meta.get("correlation").map(Value::clone_static);
+                        let http_meta = ctx.extract_meta(meta);
+                        let mut once_per_request =
+                            RequestBuilder::new(request_id, http_meta, &self.config).await?;
+                        once_per_request
+                            .append(value, meta, start, &mut self.serializer)
+                            .await?;
+                        builder = Some(once_per_request);
+                    }
+                    Some(builder) => {
+                        builder
+                            .append(value, meta, start, &mut self.serializer)
+                            .await?;
+                    }
+                }
+            }
+
+            assert!(builder.is_some(), "No builder");
+
+            let t = self
+                .config
+                .timeout
+                .map_or_else(|| Duration::from_secs(60), Duration::from_nanos);
+
+            let request = if let Some(builder) = &mut builder {
+                builder.finalize(&mut self.serializer)?;
+                builder.take_request()
+            } else {
+                error!("{ctx} Cannot convert event to azure request error.");
+                return Ok(SinkReply::FAIL);
+            };
+
+            let req_meta = extract_request_meta(&request)?;
+
+            // spawn
+            // spawn the sending task
+            tokio::task::spawn(async move {
+                // extract request meta for the response metadata from the finally prepared request
+                // the actual sent request might differ from the metadata used to create this request
+                let url = request.url();
+                if let Some(host) = url.host() {
+                    origin_uri.host = host.to_string();
+                }
+                origin_uri.port = url.port();
+                origin_uri.path = url.path().split('/').map(ToString::to_string).collect();
+                let response = client.execute_request(&request);
+
+                match timeout(t, response).await {
+                    Ok(Ok(response)) => {
+                        let response_meta = extract_response_meta(response).await?;
+                        // let headers = response.headers();
+
+                        if let Some(response_tx) = response_tx {
+                            let mut meta = task_ctx.meta(literal!({
+                                "request": std::convert::Into::<Value>::into(req_meta),
+                                "request_id": request_id.get(),
+                                // "response": std::convert::Into::<Value>::into(response_meta),
```

Comment:
the response seems interesting? Any reason why we exclude it?

Reply:
Makes sense, that was the context I was missing :) perhaps we throw in a quick comment that we always get 204's so don't need the body?

Reply:
Yes. I really should add a comment. I also need to live test correlation so there's that too ...